filebeat/input/filestream/internal/task/group.go

```diff
@@ -0,0 +1,129 @@
+package task
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+
+	"golang.org/x/sync/semaphore"
+)
+
+type Group struct {
+	limit uint64
+
+	sem *semaphore.Weighted
+	wg  *sync.WaitGroup
+
+	errs     []error
```

Comment:
After reading the definition of the `(p *Group) addErr` method, I realized that this field is a circular buffer, of max size `errSize`, for holding errors.  Might be good to mention that in comment here.  Otherwise, naively, it gives the impression that this `[]error` can grow unbounded.

Reply:
You can ignore this suggestion if you go with https://github.com/elastic/beats/pull/34906/files#r1173188613.

Reply:
Based on our conversation off-PR, you told me you're getting rid of the idea of the ring buffer for errors altogether.  So this thread is now irrelevant.